sdk/monitor/azure-monitor-opentelemetry-exporter/azure/monitor/opentelemetry/exporter/_configuration/__init__.py

```diff
@@ -182,34 +189,41 @@ def get_configuration_and_refresh_interval(self, query_dict: Optional[Dict[str,
             # No settings or version provided
             logger.warning("No settings or version provided in config response. Config not updated.")
 
+
+        notify_callbacks = False
+        current_refresh_interval = _ONE_SETTINGS_DEFAULT_REFRESH_INTERVAL_SECONDS
+        state_for_callbacks = None
+
         # Atomic state update
-        with _ConfigurationManager._state_lock:
-            latest_state = _ConfigurationManager._current_state  # Always use latest state
-            _ConfigurationManager._current_state = latest_state.with_updates(**new_state_updates)
-            return _ConfigurationManager._current_state.refresh_interval
+        with self._state_lock:
+            latest_state = self._current_state  # Always use latest state
+            self._current_state = latest_state.with_updates(**new_state_updates)
+            current_refresh_interval = self._current_state.refresh_interval
+            if 'settings_cache' in new_state_updates:
```

Comment:
If `settings_cache` is in new_state_updates, this means the fetched configurations have changed. Notify the callbacks in this case.